### DISCUSSION TITLE
withApollo Example and getServerSideProps not working!

### DISCUSSION THREAD

**Discussion by msreekm** [vercel/next.js#15214] [Help]
In  the withApollo example, there is a need  for a separate page that showcase how to use getServerSideProps , because just changing getStaticProps method in the current index.js to getServerSideProps does not work .The following changes are required to get it working with SSR-

1  update getStaticProps method in index.js  like so.. 
(change method name and remove  unstable_revalidate: 1)
export async function getServerSideProps(context) {
  
  const apolloClient = initializeApollo()

  await apolloClient.query({
    query: ALL_POSTS_QUERY,
    variables: allPostsQueryVars,
  })

  return {
    props: {
      initialApolloState: apolloClient.cache.extract(),
    },
  }
}

2. In PostList.js , 
update   if ( loading && !loadingMorePosts) return <div>Loading</div> to 
                if ( !data &&  (loading && !loadingMorePosts)) return <div>Loading</div>
this is required for page to be server rendered 

Am I doing something wrong ? 
Iam happy to create a pull request with a new page, let me know

**Comment by dukesx** [comment]
Rest seems fine but remove loading. Server rendered pages don't use loading. If you forcefully use loading from Apollo then it will lazy load the data instead of server rendering it, although the data would be in initial Apollo state but you won't see dom elements using Apollo data.
